### ISSUE TITLE
SQL Syntax error when adding a M2O field to a table with a name that is a substring of 'CREATE TABLE'

### ISSUE THREAD

**Issue by rmartin-sc** [directus/directus#10878]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

I first spotted this when I was playing around with a table named "A", but it appears to occur for any table with a name that is a substring of "CREATE TABLE".  I have tested this behaviour only on an SQLite database.

If you have a table named "A" (or, evidently, any substring of "CREATE TABLE") and you try to add a M2O field to it an SQL syntax error occurs (see below).  

I have verified this for tables named 'B', 'L', and 'ABLE'.  In each case, the '_knex_temp_alter###' part of the error message takes the place of whatever substring in "CREATE TABLE" corresponds to the table's name. 

I also checked on a table named 'PRIM' to see if the problem extends into the column definition parts of the SQL (as in a substring of 'PRIMARY KEY'), but this succeeded without error.

The problem also does not occur when adding a simple Text field.

### To Reproduce

1. Start a new instance of Directus on an empty SQLite database
2. Create a table named "A" (or any substring of "CREATE TABLE")
3. Add a M2O relational field and link the table to itself.  The key for the field does not seem to matter.
4. When you save the field, you get an SQL syntax error like the one below in which the CREATE TABLE portion of the SQL statement has the string "_knex_temp_alter###" replacing whatever substring of "CREATE TABLE" corresponds to your table name

### Errors Shown

```
{
  "errors": [
    {
      "message": "CRE_knex_temp_alter996TE TABLE `A` (`id` integer PRIMARY KEY AUTOINCREMENT NOT NULL, `another_a` integer NULL, CONSTRAINT `a_another_a_foreign` FOREIGN KEY (`another_a`) REFERENCES `A` (`id`) ON DELETE NO ACTION) - SQLITE_ERROR: near \"CRE_knex_temp_alter996TE\": syntax error",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```

### What version of Directus are you using?

9.4.2

### What version of Node.js are you using?

16.13.1

### What database are you using?

SQLite 3

### What browser are you using?

Firefox

### What operating system are you using?

macOS

### How are you deploying Directus?

Docker

**Comment by rijkvanzanten** [comment]
That very much sounds like a bug that happens in Knex, as I don't think we're doing any substring magic in Directus itself.

**Comment by nickrum** [comment]
@rmartin-sc Good find! A fix has been merged into Knex. The fact that the table name has to be a substring of `CREATE TABLE` to trigger the error really helped to fix the issue 👍

**Comment by kibertoad** [comment]
Fix was released with Knex 1.0.1.

**Comment by rijkvanzanten** [comment]
This was fixed in Knex. We'll track the update of knex to 2.0 in #10842
